datafusion/tests/sql.rs

```diff
@@ -1289,14 +1289,46 @@ async fn equijoin_implicit_syntax_reversed() -> Result<()> {
 }
 
 #[tokio::test]
-async fn cartesian_join() -> Result<()> {
-    let ctx = create_join_context("t1_id", "t2_id")?;
+async fn cross_join() -> Result<()> {
+    let mut ctx = create_join_context("t1_id", "t2_id")?;
```

Comment:
Given the logic in the cross join, I wonder if it would make sense to also test the case where multiple record batches are sent in to the build side as well as the probe side.

Reply:
Added some extra test cases for multiple partitions